maint/gen_coll.py

```diff
@@ -484,7 +484,12 @@ def dump_buffer_swap_pre():
         elif name == "reduce_scatter_block":
             G.out.append("int count = MPIR_Comm_size(comm_ptr) * recvcount;")
 
-        G.out.append("MPIR_Coll_host_buffer_alloc(sendbuf, recvbuf, count, datatype, &host_sendbuf, &host_recvbuf);")
+        if name == "reduce":
+            use_recvbuf = "comm_ptr->rank == root ? recvbuf : NULL"
```

Comment:
Don't you need to check for `root == MPI_ROOT` as well to address the inter-communicate case?
```suggestion
            use_recvbuf = "(comm_ptr->rank == root) || (MPI_ROOT == root)? recvbuf : NULL"
```

Reply:
Yes, we do. Thanks!